clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureBaseClient.groovy

```diff
@@ -122,4 +121,20 @@ public abstract class AzureBaseClient {
     responseStatusCode == HttpURLConnection.HTTP_NO_CONTENT || responseStatusCode == HttpURLConnection.HTTP_NOT_FOUND
   }
 
+  /***
+   * register the resource provider in the subscription
+   * @param resourceManagerClient - an instance of the AzureResourceManagerClient
+   */
+  void register(AzureResourceManagerClient resourceManagerClient) {
+    if (resourceManagerClient && !"".equals(providerNamespace)) {
```

Comment:
`if (resourcesManagerClient && providerNamespace) {` would probably read a little nicer and also handles the situation where providerNamespace is "" or null.